crates/isograph_config/src/compilation_options.rs

```diff
@@ -23,9 +23,64 @@ pub struct CompilerConfig {
     pub options: ConfigOptions,
 }
 
-#[derive(Default, Debug, Clone, Copy)]
+#[derive(Default, Debug, Clone)]
 pub struct ConfigOptions {
     pub on_invalid_id_type: OptionalValidationLevel,
+    pub generate_file_extensions: OptionalGenerateFileExtensions,
+}
+
+#[derive(Debug, Clone)]
+pub struct OptionalGenerateFileExtensions {
+    extensions: HashMap<String, String>,
+}
+
+impl OptionalGenerateFileExtensions {
+    pub fn get(&self, extension: &str) -> Option<&String> {
+        let mapped_extension = self.extensions.get(extension);
+
+        if mapped_extension.is_none() {
+            eprintln!(
+                "{}\nmissing .{} file extension in options.generate_file_extensions\n",
+                "Warning:".yellow(),
+                extension
+            );
+        };
+
+        mapped_extension
+    }
+    pub fn ts(&self) -> &str {
+        match self.get("ts") {
+            None => "",
+            Some(v) => v,
+        }
+    }
+    pub fn tsx(&self) -> &str {
+        match self.get("tsx") {
+            None => "",
+            Some(v) => v,
+        }
+    }
+}
+
+impl Default for OptionalGenerateFileExtensions {
+    fn default() -> Self {
+        let mut map = HashMap::new();
+        map.insert("js".to_string(), "".to_string());
```

Comment:
Instead of hardcoding this I should replace `OptionalGenerateFileExtensions` with interface, and two implementation 
- for `yes` current implementation - hashmap with warnings
- for `no` method that always returns empty string `""`